src/Sulu/Bundle/AdminBundle/Resources/js/views/Datagrid/Datagrid.js

```diff
@@ -92,11 +99,13 @@ class Datagrid extends React.Component<ViewProps> {
         router.bind('sortColumn', this.datagridStore.sortColumn);
         router.bind('sortOrder', this.datagridStore.sortOrder);
         router.bind('search', this.datagridStore.searchTerm);
+        router.bind('limit', this.datagridStore.limit);
```

Comment:
Looks like we have to add the default value here as well... otherwise the initial value gets overriden from the router (you can check that by e.g. opening the snippet form and seeing that nothing is selected in the SingleSelect if you open it).